src/components/form/HdRange.vue

```diff
@@ -0,0 +1,320 @@
+<template>
+  <div
+    :class="fieldClasses"
+    class="range field"
+  >
+    <input type="range"
+      ref="input"
+      v-model="currentValue"
+      :id="name"
+      :name="name"
+      :required="required"
+      :disabled="disabled"
+      :min="minValue"
+      :max="maxValue"
+      :step="rangeStep"
+      @input="updateRangeDecoration"
+      @change="handleChange"
+      @focus="handleFocus"
+      @blur="handleBlur"
+    >
+    <div class="range__thumb" ref="thumb">
+      <div class="range__thumb__bullet">
+      </div>
+    </div>
+    <div class="range__decoration">
+      <div class="range__progress" ref="progress">
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+
+export default {
+  name: 'HdRange',
+  props: {
+    name: {
+      type: String,
+      default: 'range',
+    },
+    required: {
+      type: Boolean,
+      default: false,
+    },
+    disabled: {
+      type: Boolean,
+      default: false,
+    },
+    minValue: {
+      type: Number,
+      default: 0,
+    },
+    maxValue: {
+      type: Number,
+      default: 100,
+    },
+    rangeStep: {
+      type: Number,
+      default: 1,
+    },
+    isLegacy: {
+      type: Boolean,
+      default: false,
+    },
+    value: {
+      type: Number,
+      default: 0,
+    },
+  },
+  data() {
+    return {
+      isActive: null,
+      currentValue: this.value,
+    };
+  },
+  computed: {
+    fieldClasses() {
+      return {
+        'field--active': this.isActive,
+        'field--disabled': this.disabled,
+      };
+    },
+  },
+  watch: {
+    minValue() {
+      this.updateRangeDecoration();
+    },
+    maxValue() {
+      this.updateRangeDecoration();
+    },
+    rangeStep() {
+      this.updateRangeDecoration();
+    },
+    value() {
+      this.currentValue = this.value;
+      this.updateRangeDecoration();
+    },
+  },
+  mounted() {
+    this.progress = this.$refs.progress;
+    this.progress.style.width = '1px';
+
+    this.thumb = this.$refs.thumb;
+
+    this.updateRangeDecoration();
+  },
+  methods: {
+    normalizeAndEmit() {
+      if (this.currentValue > this.maxValue) {
+        this.currentValue = this.maxValue;
+      } else if (this.currentValue < this.minValue) {
+        this.currentValue = this.minValue;
+      }
+
+      this.currentValue = this.currentValue - (this.currentValue % this.rangeStep);
+
+      this.$emit('input', parseFloat(this.currentValue));
+    },
+    updateRangeDecoration() {
+      this.normalizeAndEmit();
+
+      // the percentage of the value, between max and min. I.e. : 15 is the 50% between 10 and 20
+      const valuePercentage = this.currentValue / ((this.$props.maxValue - this.$props.minValue) / 100);
```

Comment:
The current implementation will work properly **only** if the minValue is `0`, this should fix it:
```suggestion
      const valuePercentage = (this.currentValue - this.minValue) / (this.maxValue - this.minValue) * 100;
```

And yeah, you don't have to use `$props`, (not like in some other frameworks, react... ehm) :D

Reply:
Supper man